test/common/config/http_subscription_test_harness.h

```diff
@@ -122,7 +122,38 @@ class HttpSubscriptionTestHarness : public SubscriptionTestHarness {
     response_json += "]}";
     envoy::api::v2::DiscoveryResponse response_pb;
     MessageUtil::loadFromJson(response_json, response_pb);
-    Http::HeaderMapPtr response_headers{new Http::TestHeaderMapImpl{{":status", "200"}}};
```

Comment:
Given the overridden function signature, we can have:

```c++
void deliverConfigUpdate(const std::vector<std::string>& cluster_names,
                           const std::string& version, bool accept) override {
	// where `modify` is true and `response_code` is 200.
    deliverConfigUpdate(cluster_names, version, accept, true, "200");
  }
```


And you need to do:

```c++
if (modify) {
      EXPECT_CALL(callbacks_, onConfigUpdate(RepeatedProtoEq(response_pb.resources()), version))
          .WillOnce(ThrowOnRejectedConfig(accept));
}
```

To control that if you don't modify, you don't expect to call the `callbacks_.onConfigUpdate`. Since you don't call `parseResponse` (that triggers the onConfigUpdate) when you decided to send `304`. https://github.com/envoyproxy/envoy/blob/6dd4b6f83119e6b284f95ed9c74e4dfcbb4efc8c/source/common/config/http_subscription_impl.cc#L83.

Reply:
How does it work, with the function being called again inside the `override {}` block, with a different function signature? 😖

Reply:
please see my comment here: https://github.com/envoyproxy/envoy/pull/6948#discussion_r284522175